### PR TITLE
Refactor XMMX keyboard to XMMX and Toad

### PR DESCRIPTION
**Pull request by mechmerlin**
I was on a hunt for a 60% with function row and sure enough I found one that fit the bill, the Toad. Unfortunately despite it being supported in QMK, I couldn't find it. Turns out it was embedded within the XMMX project.

This PR refactors the XMMX directory to pull out the Toad from it. Configurator support will come later as this was starting to look very large already.

### CODE REVIEW

File: keyboards/toad/readme.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+Toad schematic and PCB are available opensource under Creative Commons BY-SA 3.0 license on EasyEDA at [this link](https://easyeda.com/farmakon/70_Keyboard-d4f6baf4792d4ada9c0571fa3713e461)
+
+For more informations on the Toad please visit this [geekhack.org thread](https://geekhack.org/index.php?topic=91388.0)
+
+Make examples for these keyboards (after setting up your build environment):
+
+    make toad:default
+
+
+See [build environment setup](https://docs.qmk.fm/build_environment_setup.html) then the [make instructions](https://docs.qmk.fm/make_instructions.html) for more information.
```
Comment (noroadsleft) [inline keyboards/toad/readme.md]: Can you please change this to

`See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).`

File: keyboards/toad/rules.mk [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+
+# Boot Section Size in *bytes*
+OPT_DEFS += -DBOOTLOADER_SIZE=4096
+
+
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE ?= no	# Virtual DIP switch configuration(+1000)
```
Comment (noroadsleft) [inline keyboards/toad/rules.mk]: The question marks in this code block should be removed.

File: keyboards/xmmx/readme.md [hunk 3]
```diff
@@ -1,20 +1,14 @@
...
-For more informations on the Toad please visit this [geekhack.org thread](https://geekhack.org/index.php?topic=91388.0)
 
 Make examples for these keyboards (after setting up your build environment):
 
     make xmmx:default
 
-    make xmmx:toad
-
 
 See [build environment setup](https://docs.qmk.fm/build_environment_setup.html) then the [make instructions](https://docs.qmk.fm/make_instructions.html) for more information.
```
Comment (noroadsleft) [inline keyboards/xmmx/readme.md]: Could you please update this to:

`See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).`

File: keyboards/toad/config.h [hunk 4]
```diff
@@ -0,0 +1,54 @@
+#ifndef CONFIG_H
```
Comment (drashna) [inline keyboards/toad/config.h]: If you could replace the include guard (the ifndef, define, and endif at the end) with just `#pragma once`? 

It's simpler, and less prone to user error, so if you could make this switch?

File: keyboards/toad/config.h [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+#define LOCKING_RESYNC_ENABLE
+
+/* key combination for command */
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
+
+#ifdef RGB_DI_PIN
+#define RGBLIGHT_ANIMATIONS
+#define RGBLED_NUM 0
```
Comment (drashna) [inline keyboards/toad/config.h]: Why is this set do 0?

File: keyboards/toad/toad.c [hunk 6]
```diff
@@ -0,0 +1,18 @@
+#include "toad.h"
+
+void led_set_user(uint8_t usb_led) {
```
Comment (drashna) [inline keyboards/toad/toad.c]: This should be `led_set_kb`, and should call `led_set_user` at the end of it.

File: keyboards/xmmx/xmmx.c [hunk 7]
```diff
@@ -1 +1,18 @@
 #include "xmmx.h"
+
+void led_set_user(uint8_t usb_led) {
```
Comment (drashna) [inline keyboards/xmmx/xmmx.c]: This should be `led_set_kb`, and should call `led_set_user` at the end of it.

**Comment by mechmerlin** [comment]
There's a lot of work to be done here. This PR is basically just to decouple the boards from each other. Id like to make the other changes when I fix each individual board if possible along with the config changes.

**Comment by drashna** [comment]
That's fine.  I'll merge this now, and then somebody can fix it up later.

**Pull request merged**